### PR TITLE
generic parameter result

### PR DESCRIPTION
**Pull request by tanner0101**
Allows parameters to choose their own return type. This way non-async parameters can continue to provide a nice sync api.

```swift
router.get("pets", Pet.parameter, "toys") { req -> Future<[Toy]> in
    return try req.parameters.next(Pet.self).then { pet in
        return req.database(.beta) { db in
            return try pet.toys.query(on: db).all()
        }
    }
}

router.get("string", String.parameter) { req -> String in
    return try req.parameters.next(String.self)
}
```

### CODE REVIEW

File: Tests/RoutingTests/RouterTests.swift [hunk 1]
```diff
@@ -72,10 +91,6 @@ class RouterTests: XCTestCase {
     ]
 }
 
-extension Response: FutureType {
```
Comment (Joannis) [inline Tests/RoutingTests/RouterTests.swift]: Why was this removed?

Comment (tanner0101) [reply Tests/RoutingTests/RouterTests.swift]: This is covered by `Content` now. (Response conforms to Content, all Content is FutureType)
